src/test/custom_nodes/node_choose_maximum.cpp

```diff
@@ -0,0 +1,163 @@
+    std::cout << "Creating outputs:" << *outputs << std::endl;
+    // TODO check result
+    CustomNodeTensor& resultTensor = *outputs[*outputsLength - 1];
+    resultTensor.name = "maximum_tensor";
+    float* result = new float[dummyInputSize];
+    resultTensor.data = reinterpret_cast<uint8_t*>(result);
+    resultTensor.dimsLength = 2;
+    resultTensor.dims = new uint64_t[resultTensor.dimsLength];
+    resultTensor.dims[0] = 1;
+    resultTensor.dims[1] = dummyInputSize;
+    resultTensor.dataLength = 1 * dummyInputSize * sizeof(float);
+    resultTensor.precision = FP32;
+    // get input tensor
+    float* inputTensor;
+    for (size_t i = 0; i < inputsLength; ++i) {
+        if (INPUT_TENSOR_NAME == inputs[i].name) {
+            inputTensor = reinterpret_cast<float*>(inputs[i].data);
+        } else {
+            std::cout << "Lacking input:" << INPUT_TENSOR_NAME << std::endl;
+            return 1;  // TODO free
+        }
+    }
+    // get adjustable dim
+    size_t tensorsCount = inputs[0].dims[1];
+    // perform operations
+    std::vector<float> minimums(tensorsCount, std::numeric_limits<int>::max());
+    std::vector<float> maximums(tensorsCount, std::numeric_limits<int>::lowest());
+    std::vector<float> averages(tensorsCount, 0);
+    for (size_t opId = 0; opId < tensorsCount; ++opId) {  // TODO adjustable opsId
+        for (size_t dummyPos = 0; dummyPos < dummyInputSize; ++dummyPos) {
+            auto index = opId * dummyInputSize + dummyPos;
+            switch (selection_method) {
+            case METHOD::MAXIMUM_MAXIMUM:
+                maximums[opId] = std::max(maximums[opId], inputTensor[index]);
+                break;
+            case METHOD::MAXIMUM_MINIMUM:
+                minimums[opId] = std::min(maximums[opId], inputTensor[index]);
+                break;
+            case METHOD::MAXIMUM_AVERAGE:
+                averages[opId] += inputTensor[index];
+                break;
+            default:
+                return 2;
+            }
+            std::cout << "opId:" << opId
+                      << " dummyPos:" << dummyPos
+                      << " input:" << inputTensor[index]
+                      << " minimums:" << minimums[opId]
+                      << " averages:" << averages[opId]
+                      << " maximums:" << maximums[opId]
+                      << std::endl;
+        }
+        averages[opId] /= dummyInputSize;
+    }
+    // find which tensor to choose
+    size_t whichTensor = 42;
+    const std::vector<float>* fromWhichContainerToChoose = &maximums;
+    switch (selection_method) {
+    case METHOD::MAXIMUM_MAXIMUM:
+        fromWhichContainerToChoose = &maximums;
+        break;
+    case METHOD::MAXIMUM_MINIMUM:
+        fromWhichContainerToChoose = &minimums;
+        break;
+    case METHOD::MAXIMUM_AVERAGE:
+        fromWhichContainerToChoose = &averages;
+        break;
+    default:
+        // TODO cleanup of ptrs
```

Comment:
We do not need to worry about default, you already validated and returned above if parameters contained unknown method.

Reply:
I think there is no need to handle errors once you allocate outputs for this custom node case, so I would not try to make up one.